sycl/plugins/opencl/CMakeLists.txt

```diff
@@ -26,11 +27,30 @@ set_target_properties(pi_opencl PROPERTIES LINKER_LANGUAGE CXX)
 #preprocessor definitions for compiling a target's sources. We do not need it for pi_opencl
 target_include_directories(pi_opencl PRIVATE "${sycl_inc_dir}")
 
-#link pi_opencl with OpenCL headers and ICD Loader.
 target_link_libraries( pi_opencl
-    PRIVATE OpenCL::Headers
-    PRIVATE ${OpenCL_LIBRARIES}
+  PRIVATE OpenCL::Headers
+  PRIVATE ${OpenCL_LIBRARIES}
 )
+if (MSVC)
+  # by defining __SYCL_BUILD_SYCL_DLL, we can use __declspec(dllexport)
+  # which are individually tagged for all pi* symbols in pi.h
+  target_compile_definitions(pi_opencl PRIVATE __SYCL_BUILD_SYCL_DLL)
+else()
+  # we set the visibility of all symbols 'hidden' by default.
+  # In pi.h file, we set exported symbols with visibility==default individually
+  target_compile_options(pi_opencl PUBLIC -fvisibility=hidden)
+
+  # This script file is used to allow exporting pi* symbols only.
+  # All other symbols are regarded as local (hidden)
+  set(linker_script "${CMAKE_CURRENT_SOURCE_DIR}/../ld-version-script.txt")
+
+  #link pi_opencl with OpenCL headers and ICD Loader.
```

Comment:
I think this comment should stay at line 29 and comments from lines 49-50 can be moved here.

Reply:
@bader , Good catch. 👍 I fixed as you requested.